### DISCUSSION TITLE
GPU is not utilized for some tasks on Airflow

### DISCUSSION THREAD

**Discussion by ashamhusain** [apache/airflow#34832] [General]
Hello,

I am using Airflow version - 2.0.0. I have an airflow pipeline which is dynamically created. This pipeline launches and EC2 which contains various tasks among which two tasks need to use GPU on the EC2 instance. I do this by making these tasks use a DockerOperator which is configured to connect to GPUs (named DockerConfigurableOperator). It utilizes the GPU for one of the tasks and does not utilize it for another task. When I tried to run the other task independently, it still does not utilize the GPU. 

The other task where GPU is not being utilized is used for running a Segment Anything Model (from meta). I tried running the same task on EC2 instance without using Airflow and then it utilizes the GPU. Your help is appreciated on this matter.

I am attaching the code that I am using for DockerConfigurableOperator for connecting to GPUs as a txt file.
[gpu_docker_operator.txt](https://github.com/apache/airflow/files/12844563/gpu_docker_operator.txt)

**Comment by Taragolis** [comment]
> I am using Airflow version - 2.0.0

[`device_requests`](https://docker-py.readthedocs.io/en/stable/api.html) parameter was added in `apache-airflow-providers-docker==2.7.0` which supports Airflow 2.1+ and would not works in 2.0.0
- https://github.com/apache/airflow/issues/9492
- https://github.com/apache/airflow/pull/23554
- https://airflow.apache.org/docs/apache-airflow-providers-docker/stable/changelog.html#id33
